Nanoleaf-wpf/Views/Options/OptionsWindow.xaml.cs

```diff
@@ -259,5 +275,57 @@ private string FullNameForCulture(string text)
 
             return null;
         }
+
+        private void AddColor_Click(object sender, RoutedEventArgs e)
+        {
+            var color = ColorPicker.SelectedColor;
+            var name = EffectTextBox.Text;
+
+            if (UserSettings.Settings.CustomEffects != null && UserSettings.Settings.CustomEffects.Any(effect => effect.EffectName == name))
+            {
+                PopupCreator.Error(Options.Resources.NameTaken);
+                return;
+            }
+
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                PopupCreator.Error(Options.Resources.NoNameProvided);
+                return;
+            }
+
+            if (color == null)
+            {
+                PopupCreator.Error(Options.Resources.NoColorSelected);
+                return;
+            }
+
+            var customEffect = new UserCustomColorEffect()
+            {
+                Color = Color.FromArgb(color.Value.A, color.Value.R, color.Value.G, color.Value.B),
+                EffectName = name
+            };
+
+            if (UserSettings.Settings.CustomEffects == null)
+            {
+                UserSettings.Settings.CustomEffects = new List<UserCustomColorEffect>();
+            }
+
+            // Add color in settings.
+            UserSettings.Settings.CustomEffects.Add(customEffect);
+            
+            // Add color to UI.
+            ColorList.Children.Add(new ColorUserControl(this, customEffect.EffectName, customEffect.Color));
```

Comment:
When the view loads, you sort on name but here you add without sorting. Please sort after adding or remove then sorting